PracticeRobot2/src/main/java/frc/robot/Robot.java

```diff
@@ -0,0 +1,203 @@
+    SmartDashboard.putData("Auto choices", m_chooser);
+
+    // Set Encoder pulses
+    m_leftEncoder.setDistancePerPulse((0.1524 * Math.PI) / (double) 1024);
+    m_rightEncoder.setDistancePerPulse((0.1524 * Math.PI) / (double) 1024);
+    m_leftEncoder.reset();
+    m_rightEncoder.reset();
+
+    if (!RobotBase.isReal()) {
+      // Set up robot simulation
+      m_driveTrainSim = DifferentialDrivetrainSim.createKitbotSim(KitbotMotor.kDualCIMPerSide, KitbotGearing.k10p71,
+          KitbotWheelSize.SixInch, null);
+      m_fieldSim = new Field2d();
+      SmartDashboard.putData("Field", m_fieldSim);
+
+      // Connect the simulators with their counterparts
+      m_leftEncoderSim = new EncoderSim(m_leftEncoder);
+      m_rightEncoderSim = new EncoderSim(m_rightEncoder);
+      m_gyroSim = new AnalogGyroSim(m_gyro);
+    }
+  }
+
+  /**
+   * This function is called every robot packet, no matter the mode. Use this for
+   * items like diagnostics that you want ran during disabled, autonomous,
+   * teleoperated and test.
+   *
+   * <p>
+   * This runs after the mode specific periodic functions, but before LiveWindow
+   * and SmartDashboard integrated updating.
+   */
+  @Override
+  public void robotPeriodic() {
+  }
+
+  @Override
+  public void simulationPeriodic() {
+    // connect the motors to update the drivetrain
+    m_driveTrainSim.setInputs(m_leftMotor.get() * RobotController.getBatteryVoltage(),
+        m_rightMotor.get() * RobotController.getBatteryVoltage());
+
+    // Run and update simulation
+    m_driveTrainSim.update(0.02);
+    m_leftEncoderSim.setDistance(m_driveTrainSim.getLeftPositionMeters());
+    m_leftEncoderSim.setRate(m_driveTrainSim.getLeftVelocityMetersPerSecond());
+    m_rightEncoderSim.setDistance(m_driveTrainSim.getRightPositionMeters());
+    m_rightEncoderSim.setRate(m_driveTrainSim.getRightVelocityMetersPerSecond());
+    m_gyroSim.setAngle(-m_driveTrainSim.getHeading().getDegrees());
+
+    m_odometry.update(Rotation2d.fromDegrees(getHeading()), m_leftEncoder.getDistance(), m_rightEncoder.getDistance());
+    m_fieldSim.setRobotPose(getPose());
+  }
+
+  /**
+   * This autonomous (along with the chooser code above) shows how to select
+   * between different autonomous modes using the dashboard. The sendable chooser
+   * code works with the Java SmartDashboard. If you prefer the LabVIEW Dashboard,
+   * remove all of the chooser code and uncomment the getString line to get the
+   * auto name from the text box below the Gyro
+   *
+   * <p>
+   * You can add additional auto modes by adding additional comparisons to the
+   * switch structure below with additional strings. If using the SendableChooser
+   * make sure to add them to the chooser code above as well.
+   */
+  @Override
+  public void autonomousInit() {
+    m_autoSelected = m_chooser.getSelected();
+    // m_autoSelected = SmartDashboard.getString("Auto Selector", kDefaultAuto);
+    System.out.println("Auto selected: " + m_autoSelected);
```

Comment:
In order to use the timer for autonomous, you have to prep it. Add the following inside autonomousInit():

m_timer.reset();
m_timer.start();